torch/distributed/fsdp/_utils.py

```diff
@@ -2,13 +2,47 @@
 
 import torch
 from torch.nn.modules.batchnorm import _BatchNorm
+from torch.nn.modules.normalization import LayerNorm
 from torch.utils._mode_utils import no_dispatch
 
+from torch.distributed.utils import _apply_to_tensors
+from functools import partial
+
 
 def _contains_batchnorm(module):
     return any(isinstance(mod, _BatchNorm) for mod in module.modules())
 
 
+def _contains_module(root, module_to_check):
+    return any(isinstance(mod, module_to_check) for mod in root.modules())
+
+def _override_module_mixed_precision(root, module_to_override, wrap_override_dict={"mixed_precision": None}):
+    for mod in root.modules():
+        if isinstance(mod, module_to_override):
+            mod._wrap_overrides = wrap_override_dict  # type: ignore[assignment]
+
+        if isinstance(mod, LayerNorm): # TODO: generalize this logic if additional types need to be supported
```

Comment:
I think this should be replaced with `if isinstance(mod, module_to_override):`?

Reply:
I was hardcoding this to LN for now. LN would crash if it got fp16 inputs and was running in fp32, but for some reason batchnorm doesn't need this. Maybe we should just always cast up to fp32?

Reply:
Oh I see. If batch norm kernels have the casting logic, then maybe layer norm should as well? It might be good to check on why there is that difference before adding logic at the FSDP level.